sklearn/decomposition/_online_lda_fast.pyx

```diff
@@ -88,12 +117,12 @@ def _dirichlet_expectation_2d(np.ndarray[ndim=2, dtype=np.float64_t] arr):
 #
 # After: J. Bernardo (1976). Algorithm AS 103: Psi (Digamma) Function.
 # https://www.uv.es/~bernardo/1976AppStatist.pdf
-cdef double psi(double x) nogil:
+cdef float_t psi(float_t x) nogil:
```

Comment:
I think we can use cython's `floating` fused type here:

```
from cython cimport floating
```

Reply:
I have changed to use floating.